lib/resty/core/regex.lua

```diff
@@ -353,6 +353,11 @@ end
 
 
 local function re_match_helper(subj, regex, opts, ctx, want_caps, res, nth)
+    -- we need to caste these to a string to avoid a potential thread abort
+    -- see http://bit.ly/1SH0oz6
+    regex = tostring(regex)
```

Comment:
I think regexes should always be of string type. I cannot think of real use cases of passing numbers or booleans as the pattern. These cases are almost always undesired. Thoughts?